### PR TITLE
Fix #511: validate external type

### LINKED ISSUE #511: Fix external type validation algorithm
As noted [here](https://github.com/w3c/web-nfc/issues/350#issuecomment-570529776)

> Probably what we should do instead is figure out if `dns-part` is literal IPv6 and take the _first_ `:` as delimiter after it, allowing `:` in the `name-part` of external types.

### PR DESCRIPTION
**Pull request by zolkis**
Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

***
<a href="https://pr-preview.s3.amazonaws.com/zolkis/web-nfc/pull/514.html" title="Last updated on Jan 6, 2020, 10:50 AM UTC (b77c8f0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/514/b1a2853...zolkis:b77c8f0.html" title="Last updated on Jan 6, 2020, 10:50 AM UTC (b77c8f0)">Diff</a>

### CODE REVIEW

**Review by kenchris — COMMENTED** [review]
Can you explain the [] to me?

**Comment by zolkis** [comment]
> Can you explain the [] to me?

In an URL, the IPv6 address should be enclosed in "[" and "]" brackets, IIUC. [[rfc2732]](https://tools.ietf.org/html/rfc2732)
Since the NDEF spec uses this prose (referring to URIs), if this convention does not apply in your opinion in the external type RTD, then we can disallow using it in Web NFC and take out the whole step 2.

Edit: My interpretation is that even though the NFC RTD spec calls the type format RTD-URI in section 3 (RTD Type Names ABNF), but it does not require URL conventions in the `dns-part` (instead, defines its own format), so it's disputable if we should support IPv6 there.

File: index.html [hunk 1]
```diff
@@ -2812,43 +2812,33 @@ <h3><dfn>Writing or pushing content</dfn></h3>
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
-          If |input:string| is not a {{USVString}} or it is empty, or its
+          If |input:string| is not an [=ASCII string=] or it is empty, or its
```
Comment (beaufortfrancois) [inline index.html]: Nit: "If |input:string| is not an [=ASCII string=], is empty, or its length exceeds 255 bytes, return `false`."

**Review by beaufortfrancois — APPROVED** [review]
LGTM with nit

**Comment by zolkis** [comment]
Okay, this became a very simple update, in the end.
As discussed in #511, we don't need to support IPv6 in the external type, since domain name is mainly used for namespacing and we don't expect relationship between the `dns-name` part used in external types and the document origin.

File: index.html (hunk 1 above)
Comment (zolkis) [reply index.html]: Fixed, thanks.

**Comment by zolkis** [comment]
Merging with keeping both commits.

**Pull request merged**